### PR TITLE
fix(many): disabled control in item does not receive active/hover states

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

`ion-checkbox`, `ion-toggle`, `ion-select` and `ion-radio` apply their active/hover styles when they are disabled inside of an `ion-item`.

e.g.:
```html
<ion-item>
  <ion-checkbox disabled="true"></ion-checkbox>
</ion-item>
```

Issue URL: #26706

## What is the new behavior?

- Active and hover styles are not applied when `ion-checkbox`, `ion-toggle`, `ion-select` or `ion-radio` are disabled
- Refactored implementations to have a common implementation pattern (e.g. using `protected` for watch callbacks)

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by amandaejohnston — COMMENTED** [review]
Could we add this to the test pages somewhere so it's easy to check in the future?

File: core/src/components/toggle/toggle.tsx [hunk 1]
```diff
@@ -137,10 +137,11 @@ export class Toggle implements ComponentInterface {
   @Event() ionStyle!: EventEmitter<StyleEventDetail>;
 
   @Watch('disabled')
-  disabledChanged() {
+  protected disabledChanged() {
     this.emitStyle();
-    if (this.gesture) {
```
Comment (liamdebeasi) [inline core/src/components/toggle/toggle.tsx:142]: Do we need this fix here? I thought toggle doesn't currently receive hover/active states from item.

Comment (sean-perkins) [reply core/src/components/toggle/toggle.tsx:142]: 🤔 doesn't look like it. I could of sworn I tested each of these in the browser first to figure out which ones needed to be adjusted and the hover styles were applying, but maybe I was mistaken with another component. 

I'll revert all the changes to this file.

**Comment by amandaejohnston** [comment]
It doesn't look like any of the newly added tests have the controls disabled? The screenshots don't look like they have hover styles on them anyway, though, so I'm guessing Playwright's gesture support is insufficient here. If that's the case, I wouldn't be opposed to just having some disabled controls in items in one of the HTML pages somewhere, so we can check it manually. Maybe with a tech debt ticket to revisit in the future.

**Comment by sean-perkins** [comment]
This should be ready for another review 👍

**Review by liamdebeasi — APPROVED** [review]

**Pull request merged**